src/main/java/org/spongepowered/common/mixin/core/entity/living/MixinEntityLiving.java

```diff
@@ -94,7 +93,7 @@ public void setCanPickupItems(boolean canPickupItems) {
 
     @Inject(method = "interactFirst", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/EntityLiving;setLeashedToEntity(Lnet/minecraft/entity/Entity;Z)V"), locals = LocalCapture.CAPTURE_FAILEXCEPTION, cancellable = true)
     public void callLeashEvent(EntityPlayer playerIn, CallbackInfoReturnable<Boolean> ci, ItemStack itemstack) {
-        final EntityLeashEvent event = SpongeEventFactory.createEntityLeash(Sponge.getGame(), this, (Player)playerIn);
+        final EntityLeashEvent event = SpongeEventFactory.createEntityLeash(Sponge.getGame(), this, (Entity)playerIn);
```

Comment:
@Zidane Do you think this part could be merged? It shouldn't affect anything, except for fixing the crash.